src/Nemo.jl

```diff
@@ -251,7 +251,16 @@ function __init__()
    ccall((:flint_set_abort, libflint), Nothing,
          (Ptr{Nothing},), @cfunction(flint_abort, Nothing, ()))
 
-   if isinteractive() &&
+   # Check if were are non-interactive
+   io = IOBuffer()
+   Base.show_backtrace(io, Base.backtrace())
+   bt = String(take!(io))
+   isinteractive_manual = !occursin("_tryrequire_from_serialized", bt)
```

Comment:
I didn't check what `_tryrequire_from_serialized` does, but that's a nice trick! Maybe expand a bit the comment above to explain the difference from `isinteractive()` ? Also, doesn't this supersedes the line with `!any(x -> x.name in ("Hecke", "Oscar", "Singular"), keys(Base.package_locks))` ?